### DISCUSSION TITLE
NO DOCKER: hydra serve all --config  hydra-sqlite-sample.yml ?

### DISCUSSION THREAD

**Discussion by anborg** [ory/hydra#2286] [Q&A] [answered]
hydra serve all --config  **hydra-sqlite-sample.yml**

I want to start hydra in command line. NO DOCKER! Only commandline, sqlite or inmem.

- using sqlite file
- using some sample pem file. for TLS/https
- e.g everything just relying on localhost. 
- gmail & github as identity providers
- sqlite to be used for userid/pwd based authentication

Is there a sample config file that I can use as starter?

**Comment by anborg** [comment]
I tried using the config file in test folder https://github.com/ory/hydra/blob/master/test/conformance/hydra/config/hydra.yml

**hydra serve all --config  test/conformance/hydra/config/hydra.yml** 

{"audience":"application","level":"fatal","msg":"dsn must be set","service_name":"ORY Hydra","service_version":"v1.9.0-alpha.3","time":"2021-01-09T13:51:59-05:00"}

Would be great, if I have a default localhost hydra config, with sqlite

**Comment by k9ert** [comment, marked as answer]
To get a similiar experience of the current (v1.8.5)  quickstart without using docker, you can start it like this:
```
export URLS_SELF_ISSUER=http://127.0.0.1:4444
export URLS_CONSENT=http://127.0.0.1:3000/consent
export URLS_LOGIN=http://127.0.0.1:3000/login
export URLS_LOGOUT=http://127.0.0.1:3000/logout
export DSN=memory
export SECRETS_SYSTEM=youReallyNeedToChangeThis
export OIDC_SUBJECT_IDENTIFIERS_SUPPORTED_TYPES=public,pairwise
export OIDC_SUBJECT_IDENTIFIERS_PAIRWISE_SALT=youReallyNeedToChangeThis
export SERVE_COOKIES_SAME_SITE_MODE=Lax
hydra serve all --dangerous-force-http 
```
This will use memory. I'll have a look in your other issues as well but maybe that helps at least as a starting point.

**Reply by ganeshkbhat** [reply]
I am using Windows machine. Any recommendations here on the command line I need to add for `hydra migrate sql` and `hydra serve all` commands?

**Comment by vinckr** [comment]
Hey @anborg, just checking in if your problem still persists. 
If not, could you please select the correct answer or provide your solution to mark this as answered?
Thanks a ton 🙏

**Reply by vinckr** [reply]
Since the topic seems to be stale, I will be selecting the answer for now. 

If you run into the same issue, feel free to respond in this discussion

**Reply by anborg** [reply]
Hi @vinckr 

Thanks for the info. 

The "memory" option won't allow to persist e.g data.sqlite

I'm looking for a way to persist sqlite data in a file.

**Comment by zepatrik** [comment]
If you would have looked into the docker-compose `quickstart.yml` you would have noted that the config file is here: https://github.com/ory/hydra/blob/1c1433ab3ae0b3b50a0f911bb4a909cb4ff774fb/contrib/quickstart/5-min/hydra.yml
To use sqlite with a file you can use (also from the `quickstart.yml`):

```
export DSN=sqlite:///path/to/db.sqlite?_fk=true
```

or set it in the config file.

You will also see in the `quickstart.yml` that you need to run migrations first, that is left as an exercise for the reader :wink:

**Reply by ganeshkbhat** [reply]
I want to try this in Windows machine and this will not work. Any suggestions?

I am using Windows machine. Any recommendations here on the command line I need to add for `hydra migrate sql` and `hydra serve all` commands?

**Reply by Benehiko** [reply]
@ganeshkbhat maybe add the sqlite connection string to the hydra yaml file 
`dsn: sqlite:///path/to/db.sqlite?_fk=true` and then use the config files (hydra migrate sql -c /path/to/config.yml) instead of the environment variable
https://www.ory.sh/hydra/docs/cli/hydra-migrate-sql
